Sources/PrimerSDK/Classes/User Interface/OAuth/OAuthViewController.swift

```diff
@@ -238,46 +271,49 @@ class WebViewController: UIViewController, WKNavigationDelegate {
         ]
 
         if let url = navigationAction.request.url, let host = url.host, allowedHosts.contains(host) {
-
             let val = url.queryParameterValue(for: "token")
+            
+            if (val ?? "").isEmpty || val == "undefined" {
+                let err = PrimerError.clientTokenNull
+                klarnaWebViewCompletion?(nil, err)
+                decisionHandler(.cancel)
+                return
+            }
 
             log(logLevel: .info, message: "🚀🚀 \(url)")
             log(logLevel: .info, message: "🚀🚀 token \(val)")
             
             let state: AppStateProtocol = DependencyContainer.resolve()
 
+            // FIXME: WebView should be agnostic of state and not set it. Debug what happens on
```

Comment:
one problem we have at the moment is indeed global navigation state. IMO, the view models should update a central context store with some sort of callback to a business logic layer that's completely unknown to the view models/ view controllers. this business logic layer should notify the global navigation state which in turn determines what view to display/instantiate.